configure.ac

```diff
@@ -2400,7 +2410,15 @@ CheckGLX()
         AC_MSG_RESULT($video_opengl_glx)
         if test x$video_opengl_glx = xyes; then
             AC_DEFINE(SDL_VIDEO_OPENGL, 1, [ ])
-            AC_DEFINE(SDL_VIDEO_OPENGL_GLX, 1, [ ])
+            #
+            # Disable GLX on ROCKCHIP platform with mali blob.
+            # This make sure SDL loads OpenGLES2 with HW acceleration
+            #
+            if test x$enable_video_rockchip = xyes; then
+                AC_MSG_WARN([GLX disabled for ROCKCHIP HW acceleration])
+            else
+                AC_DEFINE(SDL_VIDEO_OPENGL_GLX, 1, [ ])
+            fi
```

Comment:
There is probably a cleaner way to do this.

Maybe you could default `SDL_HINT_VIDEO_X11_FORCE_EGL` to `1` on Rockchip builds? That would at least allow apps that _really_ want GLX to get it.

Reply:
I will revise the code, but if I recall correctly if SDL_VIDEO_OPENGL_GLX = 1, SDL does not make use of libegles2.so.

Reply:
Maybe it was here:

```
    /* If the profile requested is not GL ES, switch over to X11_GL functions  */
    if ((_this->gl_config.profile_mask != SDL_GL_CONTEXT_PROFILE_ES) &&
        !SDL_GetHintBoolean(SDL_HINT_VIDEO_X11_FORCE_EGL, SDL_FALSE)) {
        #if SDL_VIDEO_OPENGL_GLX
        X11_GLES_UnloadLibrary(_this);
        _this->GL_LoadLibrary = X11_GL_LoadLibrary;
        _this->GL_GetProcAddress = X11_GL_GetProcAddress;
        _this->GL_UnloadLibrary = X11_GL_UnloadLibrary;
        _this->GL_CreateContext = X11_GL_CreateContext;
        _this->GL_MakeCurrent = X11_GL_MakeCurrent;
        _this->GL_SetSwapInterval = X11_GL_SetSwapInterval;
        _this->GL_GetSwapInterval = X11_GL_GetSwapInterval;
        _this->GL_SwapWindow = X11_GL_SwapWindow;
        _this->GL_DeleteContext = X11_GL_DeleteContext;
        return X11_GL_LoadLibrary(_this, path);
        #else
        return SDL_SetError("SDL not configured with OpenGL/GLX support");
        #endif
    }

```

Reply:
I will test with SDL_HINT_VIDEO_X11_FORCE_EGL = 1

Reply:
I think it has something to do with:

#if SDL_VIDEO_OPENGL
    _this->gl_config.major_version = 2;
    _this->gl_config.minor_version = 1;
    _this->gl_config.profile_mask = 0;
#elif SDL_VIDEO_OPENGL_ES2
    _this->gl_config.major_version = 2;
    _this->gl_config.minor_version = 0;
    _this->gl_config.profile_mask = SDL_GL_CONTEXT_PROFILE_ES;
#elif SDL_VIDEO_OPENGL_ES
    _this->gl_config.major_version = 1;
    _this->gl_config.minor_version = 1;
    _this->gl_config.profile_mask = SDL_GL_CONTEXT_PROFILE_ES;
#endif

my SDL_config.h

/* #undef SDL_VIDEO_RENDER_D3D */
/* #undef SDL_VIDEO_RENDER_D3D11 */
#define SDL_VIDEO_RENDER_OGL 1
/* #undef SDL_VIDEO_RENDER_OGL_ES */
#define SDL_VIDEO_RENDER_OGL_ES2 1
/* #undef SDL_VIDEO_RENDER_DIRECTFB */
/* #undef SDL_VIDEO_RENDER_METAL */

/* Enable OpenGL support */
#define SDL_VIDEO_OPENGL 1
/* #undef SDL_VIDEO_OPENGL_ES */
#define SDL_VIDEO_OPENGL_ES2 1
/* #undef SDL_VIDEO_OPENGL_BGL */
/* #undef SDL_VIDEO_OPENGL_CGL */
#define SDL_VIDEO_OPENGL_EGL 1
/* #undef SDL_VIDEO_OPENGL_GLX */
/* #undef SDL_VIDEO_OPENGL_WGL */
/* #undef SDL_VIDEO_OPENGL_OSMESA */
/* #undef SDL_VIDEO_OPENGL_OSMESA_DYNAMIC */

Reply:
Anyway, i fix this when tested.

Reply:
Unfortunately according to some documentation and forums, mail does not support GLX only GLES.
Enabling SDL_VIDEO_OPENGL_GLX will revert to Software Rendering and not use mali.

Reply:
https://developer.arm.com/documentation/dui0363/d/introduction/graphics-standards

In this case, we must not set SDL_VIDEO_OPENGL_GLX to 1 if one wants hw 3D accel.
If the user wants GLX it should not pass --enable-video-rockchip in the command line configuration.